CMakeLists.txt

```diff
@@ -112,7 +112,7 @@ target_include_directories(${LIBRARY_TARGET_NAME} PUBLIC "$<BUILD_INTERFACE:${CM
 
 ADD_WARNINGS_CONFIGURATION_TO_TARGETS(PRIVATE TARGETS ${LIBRARY_TARGET_NAME})
 
-target_link_libraries(${LIBRARY_TARGET_NAME} PUBLIC osqp::osqp Eigen3::Eigen)
+target_link_libraries(${LIBRARY_TARGET_NAME} PUBLIC ${OSQP_EIGEN_OSQP_TARGET_TO_LINK} Eigen3::Eigen)
```

Comment:
This is just some syntactic sugar, but it might be more intuitive for `OSQP_EIGEN_OSQP_TARGET_TO_LINK:STRING` to instead be something like `OSQP_EIGEN_LINK_OSQP_SHARED:STRING="TRUE"|"FALSE"|""` since there are only two possible OSQP library targets and a default case. 

This is just a suggestion and is not a huge deal either way.

Reply:
The problem is that if you are using osqp v0.6.3 (or a osqp patched to observe BUILD_SHARED_LIBS, that we had internally for some projects) `osqp::osqp` is not automatically the shared lib, so I do not really like the `OSQP_EIGEN_LINK_OSQP_SHARED` variable name. Explicitly passing the target to link seems to me less ambiguous, and as it is anyhow an expert options that normal users would not use, the fact that you can pass a non-existent target is not a big concern to me.